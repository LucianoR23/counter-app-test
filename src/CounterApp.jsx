import { useState } from 'react'
import PropTypes from 'prop-types'

export const CounterApp = ({ value }) => {

    const [ counter, setCounter ] = useState( value )

    const handleAdd = () => {
        // setCounter( counter + 1 )
        setCounter( (c) => c + 1 )
    }
    const handleSubstract = () => {
        setCounter( counter - 1 )
    }
    const handleReset = () => {
        setCounter( value )
    }

    return (
    <>
        <h1>CounterApp</h1>
        <h2>{ counter }</h2>
        <button className='btn btn-secondary' onClick={ handleSubstract }>-1</button>
        <button className='btn btn-danger' onClick={ handleReset }>Reset</button>
        <button className='btn btn-primary' onClick={ handleAdd }>+1</button>
    </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number,
}