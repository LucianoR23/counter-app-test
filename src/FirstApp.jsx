import PropTypes from 'prop-types'

export const FirstApp = ( {title, subtitle, name} ) => {
    return (
    <>
        <h1 data-testid="test-title" > {title} </h1>
        <h2>{subtitle}</h2>
        <h2>{subtitle}</h2>
        <p>{name}</p>
    </>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string
}

FirstApp.defaultProps = {
    // title: 'No hay titulo',
    subtitle: 'No hay subtitulo',
    name: 'No hay nombre'
}