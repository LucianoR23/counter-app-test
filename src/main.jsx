import React from 'react';
import ReactDOM from 'react-dom';
import { FirstApp } from './FirstApp';
import 'bootstrap/dist/css/bootstrap.css'
import './styles.css'
import { CounterApp } from './CounterApp';


ReactDOM.createRoot( document.getElementById('root') ).render(
    <React.StrictMode>
        <CounterApp value={0} />
    </React.StrictMode>
)