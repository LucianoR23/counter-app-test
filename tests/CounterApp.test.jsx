/**
 * @jest-environment jsdom
 */

import { fireEvent, render, screen } from "@testing-library/react"
import { CounterApp } from "../src/CounterApp"


describe('Pruebas en el CounterApp', () => {
    
    const initValue = 0

    test('debe de hacer match con el snapshot', () => {

        const { container } = render( <CounterApp value={ initValue }/>)
        expect( container ).toMatchSnapshot()

    })

    test('Debe mostrar el valor inicial de 100', () => {

        render( <CounterApp value={ 100 }/>)
        expect( screen.getByText(100) ).toBeTruthy()

    })

    test('Debe incrementar con el boton +1', () => {

        render( <CounterApp value={ initValue }/> )
        fireEvent.click( screen.getByRole('button', { name: 'btn-sumar' }) )
        expect( screen.getByText('1') ).toBeTruthy()

    })

    test('Debe incrementar con el boton -1', () => {

        render( <CounterApp value={ 2 }/> )
        fireEvent.click( screen.getByRole('button', { name: 'btn-restar' }) )
        expect( screen.getByText('1') ).toBeTruthy()

    })

    test('debe de funcionar el boton de reset', () => {

        render( <CounterApp value={ initValue }/>)
        fireEvent.click( screen.getByRole('button', { name: 'btn-sumar' }) )
        fireEvent.click( screen.getByRole('button', { name: 'btn-sumar' }) )
        fireEvent.click( screen.getByRole('button', { name: 'btn-reset' }) )
        expect( screen.getByText('0') ).toBeTruthy()

    })

})