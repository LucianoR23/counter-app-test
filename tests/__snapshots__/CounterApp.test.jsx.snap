// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el CounterApp debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    0
  </h2>
  <button
    aria-label="btn-restar"
    class="btn btn-secondary"
  >
    -1
  </button>
  <button
    aria-label="btn-reset"
    class="btn btn-danger"
  >
    Reset
  </button>
  <button
    aria-label="btn-sumar"
    class="btn btn-primary"
  >
    +1
  </button>
</div>
`;
